From 1772c3518f43bbceedbb60e570676374e6263957 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Mon, 6 Apr 2020 12:31:13 +0530
Subject: [PATCH 0381/1921] octeontx2-pf: Remove deprecated skb_frag_struct

This patch get's rid of using deprecared skb_frag_struct
structure and makes use of skb_frag_t.

Change-Id: I724c10689208e7fb63134c3946531b46237b9d7b
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 .../net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c    | 2 +-
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c   | 9 ++++-----
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
index 7f51893b14d7..a98f8d12b442 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
@@ -45,7 +45,7 @@ struct otx2_stat {
 }
 
 #define OTX2_ETHTOOL_SUPPORTED_MODES 0x638CE23 //110001110001100111000100011
-#define OTX2_ETHTOOL_ALL_MODES (BIT_ULL(__ETHTOOL_LINK_MODE_LAST) - 1)
+#define OTX2_ETHTOOL_ALL_MODES (BIT_ULL(ETHTOOL_LINK_MODE_FEC_BASER_BIT) - 1)
 
 static const struct otx2_stat otx2_dev_stats[] = {
 	OTX2_DEV_STAT(rx_bytes),
diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c
index c3d4dcd2aefb..be8899916f7b 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c
@@ -46,7 +46,7 @@ static inline unsigned int frag_num(unsigned int i)
 static dma_addr_t otx2_dma_map_skb_frag(struct otx2_nic *pfvf,
 					struct sk_buff *skb, int seg, int *len)
 {
-	const struct skb_frag_struct *frag;
+	const skb_frag_t *frag;
 	struct page *page;
 	int offset;
 
@@ -58,7 +58,7 @@ static dma_addr_t otx2_dma_map_skb_frag(struct otx2_nic *pfvf,
 	} else {
 		frag = &skb_shinfo(skb)->frags[seg - 1];
 		page = skb_frag_page(frag);
-		offset = frag->page_offset;
+		offset = skb_frag_off(frag);
 		*len = skb_frag_size(frag);
 	}
 	return otx2_dma_map_page(pfvf, page, offset, *len,
@@ -688,7 +688,7 @@ static u64 otx2_tso_frag_dma_addr(struct otx2_snd_queue *sq,
 				  struct sk_buff *skb, int seg,
 				  u64 seg_addr, int hdr_len, int sqe)
 {
-	const struct skb_frag_struct *frag;
+	const skb_frag_t *frag;
 	struct sg_list *sg = &sq->sg[sqe];
 	int offset;
 
@@ -696,8 +696,7 @@ static u64 otx2_tso_frag_dma_addr(struct otx2_snd_queue *sq,
 		return sg->dma_addr[0] + (seg_addr - (u64)skb->data);
 
 	frag = &skb_shinfo(skb)->frags[seg];
-	offset = (u64)(page_address(frag->page.p) + frag->page_offset);
-	offset = seg_addr - offset;
+	offset = seg_addr - (u64)skb_frag_address(frag);
 	if (skb_headlen(skb) - hdr_len)
 		seg++;
 	return sg->dma_addr[seg] + offset;
-- 
2.31.1

